packages/cli/internal/pkg/storage/option_file_client.go

```diff
@@ -0,0 +1,123 @@
+package storage
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/aws/amazon-genomics-cli/internal/pkg/aws/s3"
+	"github.com/aws/amazon-genomics-cli/internal/pkg/cli/clierror/actionableerror"
+	"github.com/rs/zerolog/log"
+)
+
+type OptionInstance struct {
+	S3 s3.Interface
+}
+
+func NewOptionClient(S3 s3.Interface) *OptionInstance {
+	return &OptionInstance{S3}
+}
+
+func (oc *OptionInstance) UpdateOptionReferenceAndUploadToS3(initialProjectDirectory string, tempProjectDirectory string, bucketName string, baseS3Key string) error {
+	if !DoesManifestExistInDirectory(tempProjectDirectory) {
+		log.Debug().Msgf("No Manifest file was found in folder %s", tempProjectDirectory)
+		return nil
+	}
+
+	manifest, err := ReadManifestInDirectory(tempProjectDirectory)
+	if err != nil {
+		return err
+	}
+
+	fileLocation := fmt.Sprintf("%s/%s", tempProjectDirectory, manifest.OptionFileUrl)
+	optionsReferenceFile, err := ioutilReadFile(fileLocation)
+	if err != nil {
+		return err
+	}
+
+	var optionFile interface{}
+	err = jsonUnmarshall(optionsReferenceFile, &optionFile)
+	if err != nil {
+		return actionableerror.New(err, fmt.Sprintf("Please validate that the options JSON file %s exists", manifest.OptionFileUrl))
+	}
+
+	_, err = oc.UpdateOptionFile(initialProjectDirectory, optionFile, bucketName, baseS3Key, fileLocation)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (oc *OptionInstance) UpdateOptionFile(initialProjectDirectory string, optionFile interface{}, bucketName string, baseS3Key string, fileLocation string) (interface{}, error) {
+	var updatedOptionReferenceFile interface{}
+	var optionReference string
+	switch typedValue := optionFile.(type) {
+	case string:
+		updatedReference, err := oc.uploadReferenceToS3(typedValue, initialProjectDirectory, bucketName, baseS3Key)
```

Comment:
Do we have files/urls as keys in the optionsFile? I though it was all values/params for the engine.